nc/include/Entity.h

```diff
@@ -43,7 +43,7 @@ namespace nc
             template<std::derived_from<Component> T>
             bool RemoveUserComponent() noexcept;
 
-            const std::vector<std::unique_ptr<Component>> & GetUserComponents() const noexcept;
+            const std::vector<std::unique_ptr<Component>>& GetUserComponents() const noexcept;
```

Comment:
It feels wrong to have unique_ptr in the return type here. Since the performance of this method is not important, I'm thinking of constructing a vector<Component*> to return on each call.